tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,723 @@
+    """Build fixture from MockTrainer"""
+    return MockTrainer()
+
+
+@pytest.fixture
+def dummy_sample() -> DummySample:
+    """Build fixture from DummySample"""
+    return DummySample()
+
+
+def test_default_trainer_step_initiator() -> None:
+    """Test constructor of DefaultTrainerStep component"""
+    trainer_step = DefaultTrainerStep()
+    assert trainer_step.config.random_key == 42
+
+
+def test_on_training_step_with_timestamp(mock_trainer: Trainer) -> None:
+    """Test on_training_step method from TrainerStep case of existing timestamp"""
+    trainer_step = DefaultTrainerStep()
+    old_timestamp = mock_trainer.store.timestamp
+    trainer_step.on_training_step(trainer=mock_trainer)
+
+    assert int(mock_trainer.store.timestamp) == int(time.time())
+
+    assert list(mock_trainer.store.trainer_parameter_client.params.keys()) == [
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_trainer.store.trainer_parameter_client.params["trainer_steps"] == 1
+    elapsed_time = int(time.time() - old_timestamp)
+    assert (
+        int(mock_trainer.store.trainer_parameter_client.params["trainer_walltime"])
+        == elapsed_time
+    )
+
+    assert mock_trainer.store.trainer_parameter_client.call_set_and_get_async == True
+
+    assert mock_trainer.store.trainer_logger.written == {"next_sample": 2, "sample": 1}
+
+
+def test_on_training_step_without_timestamp(mock_trainer: Trainer) -> None:
+    """Test on_training_step method from TrainerStep case of no timestamp"""
+    trainer_step = DefaultTrainerStep()
+    del mock_trainer.store.timestamp
+    trainer_step.on_training_step(trainer=mock_trainer)
+
+    assert int(mock_trainer.store.timestamp) == int(time.time())
+
+    assert list(mock_trainer.store.trainer_parameter_client.params.keys()) == [
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_trainer.store.trainer_parameter_client.params["trainer_steps"] == 1
+    assert (
+        int(mock_trainer.store.trainer_parameter_client.params["trainer_walltime"]) == 0
+    )
+
+    assert mock_trainer.store.trainer_parameter_client.call_set_and_get_async == True
+
+    assert mock_trainer.store.trainer_logger.written == {"next_sample": 2, "sample": 1}
+
+
+def test_mapg_with_trust_region_step_initiator() -> None:
+    """Test constructor of MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    assert mapg_with_trust_region_step.config.discount == 0.99
+
+
+def test_on_training_init_start(mock_trainer: MockTrainer) -> None:
+    """Test on_training_init_start method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    mapg_with_trust_region_step.on_training_init_start(trainer=mock_trainer)
+
+    assert (
+        mock_trainer.store.full_batch_size
+        == mock_trainer.store.sample_batch_size
+        * (mock_trainer.store.sequence_length - 1)
+    )
```

Comment:
Can we please set this to an expected value? Because it seems like it using the code as used in the component